AlpsSDK/ProximityHandler.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  ProximityHandler.swift
+//  AlpsSDK
+//
+//  Created by Wen on 26.10.17.
+//  Copyright © 2017 Alps. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+import Alps
+
+protocol ProximityHandlerDelegate : class {
+    func didRangeBeacons(manager: ContextManager, beacons: [CLBeacon], knownBeacons: [IBeaconTriple])
+}
+
+class ProximityHandler: NSObject, ProximityHandlerDelegate {
+    var refreshTimer: Int = 60 * 1000 // timer is in milliseconds
+    var beaconsDetected: [String: [IBeaconTriple]] = [:]
+    var beaconsTriggered: [String: [String: ProximityEvent]] = [:]
+    
+    override init() {
+        super.init()
+        beaconsTriggered["unknown"] = [String: ProximityEvent]()
+        beaconsTriggered["immediate"] = [String: ProximityEvent]()
+        beaconsTriggered["far"] = [String: ProximityEvent]()
+        beaconsTriggered["near"] = [String: ProximityEvent]()
+    }
+    
+    func didRangeBeacons(manager: ContextManager, beacons: [CLBeacon], knownBeacons: [IBeaconTriple]) {
+        let unknownBeacons = beacons.filter {$0.proximity == CLProximity.unknown}
```

Comment:
it looks like all these variables could be lazy calculated class's lets. For example:
```
let unknownBeacons {
  return beacons.filter {$0.proximity == CLProximity.unknown}
}
```

Reply:
I can research for `@objc`, it is still a bit confusing for me, I didn't use objective-c. Will have a look at it